static/src/javascripts/projects/common/modules/identity/email-preferences.js

```diff
@@ -0,0 +1,122 @@
+                            subscriptionState = false;
+                        } else {
+                            subscriptionState = response.subscriptions[0].subscribedTo;
+                        }
+                    } catch (err) {
+                        renderErrorMessage(buttonEl);
+                    } finally {
+                        updateButton(buttonEl, subscriptionState);
+                    }
+                }
+            });
+        });
+    }
+
+    function enhanceEmailPreferences() {
+        $.forEachElement('.email-subscription__button', useReqwest);
+    }
+
+    function encodeFormData(csrfToken, buttonVal, htmlPreference) {
+        return 'csrfToken=' + encodeURIComponent(csrfToken) + '&' +
+        'addEmailSubscription=' + encodeURIComponent(buttonVal) + '&' +
+        'htmlPreference=' + encodeURIComponent(htmlPreference);
+    }
+
+    function renderErrorMessage(buttonEl) {
+        // appends an error message on the parent div of the button
+        var errorMessage = 'Sorry, an error has occurred, please refresh the page and try again';
+        return fastdom.write(function () {
+            var insertionPoint = $.ancestor(buttonEl, 'email-subscription');
+            // Only append an error message once for each email subscription DIV
+            if (qwery('.form__error', insertionPoint).length < 1) {
+                var errorMessageDiv = document.createElement('div');
+                errorMessageDiv.innerHTML = errorMessage;
+                errorMessageDiv.classList.add('form__error');
+                insertionPoint.appendChild(errorMessageDiv);
+            }
+        });
+    }
+
+    function updateButton(buttonEl, subscriptionState) {
+        // do this after successful POST request to update the appearance and value of the button
+        var buttonVal = buttonEl.value;
+        if (subscriptionState === true) {
+            fastdom.write(function () {
+                buttonEl.value = 'unsubscribe-' + buttonVal;
+                buttonEl.innerHTML = 'Unsubscribe';
+                $.ancestor(buttonEl, 'email-subscription').classList.add('email-subscription--subscribed');
+                buttonEl.disabled = false;
+            });
+        } else if (subscriptionState === false) {
+            fastdom.write(function () {
+                buttonEl.value = buttonVal.replace('unsubscribe-', '');
+                buttonEl.innerHTML = 'Subscribe';
+                $.ancestor(buttonEl, 'email-subscription').classList.remove('email-subscription--subscribed');
+                buttonEl.disabled = false;
+            });
+        } else {
+            renderErrorMessage(buttonEl);
+        }
+    }
+
+    function getHTMLPref() {
+        if(qwery('#htmlPreference_HTML')[0].checked) {
+          return 'HTML';
+      } else if(qwery('#htmlPreference_Text')[0].checked) {
+          return 'Text';
+      } else {
+          return '';
```

Comment:
I didn't think it could, until I got an error when testing the page and tracked it down to the data missing from the radio buttons... 😦 So added this for the weird edge case I seemed to run across